test/integration/targets/ecs_tag/tasks/full_test.yml

```diff
@@ -0,0 +1,350 @@
+      ignore_errors: yes
+
+    - name: cluster tags - check that there was no fail, but changed is false
+      assert:
+        that:
+          - taglist.failed == false
+          - taglist.changed == false
+
+
+    - name: cluster tags - invalid cluster name
+      ecs_tag:
+        cluster_name: "{{resource_prefix}}-foo"
+        resource: "{{resource_prefix}}-foo"
+        resource_type: cluster
+        state: absent
+        tags:
+          temp:
+          temp_two:
+        <<: *aws_connection_info
+      register: taglist
+      ignore_errors: yes
+
+    - name: cluster tags - Make sure invalid clustername is handled
+      assert:
+        that:
+          - taglist.failed == true
+          - taglist.changed == false
+    # Test tagging service resource
+
+    - name: services tags - Add name tag
+      ecs_tag:
+        cluster_name: "{{resource_prefix}}"
+        resource: "{{ecs_service_creation.service.serviceName}}"
+        resource_type: service
+        state: present
+        tags:
+          Name: "service-{{resource_prefix}}"
+        <<: *aws_connection_info
+      register: taglist
+
+    - name: service tag - name tags should be there
+      assert:
+        that:
+          - taglist.changed == true
+          - taglist.added_tags.Name == "service-{{ resource_prefix }}"
+
+    - name: service tags - retrieve all tags on a service
+      ecs_tag:
+        cluster_name: "{{resource_prefix}}"
+        resource: "{{ecs_service_creation.service.serviceName}}"
+        resource_type: service
+        state: list
+        <<: *aws_connection_info
+      register: taglist
+
+    - name: services tags - should have 1 tag
+      assert:
+        that:
+          - taglist.tags|list|length == 1
+          - taglist.failed == false
+          - taglist.changed == false
+
+    - name: service tags - remove service tags
+      ecs_tag:
+        cluster_name: "{{resource_prefix}}"
+        resource: "{{ecs_service_creation.service.serviceName}}"
+        resource_type: service
+        state: absent
+        tags:
+          Name:
+        <<: *aws_connection_info
+      register: taglist
+
+    - name: service tags - all tags gone
+      assert:
+        that:
+          - taglist.tags|list|length == 0
+          - taglist.changed == true
+          - '"Name" not in taglist.tags'
+
+
+    # Test tagging task_definition resource
+
+    - name: task_definition tags - Add name tag
+      ecs_tag:
+        cluster_name: "{{resource_prefix}}"
+        resource: "{{ecs_taskdefinition_creation.taskdefinition.family}}"
+        resource_type: task_definition
+        state: present
+        tags:
+          Name: "task_definition-{{resource_prefix}}"
+        <<: *aws_connection_info
+      register: taglist
+
+    - name: task_definition tag - name tags should be there
+      assert:
+        that:
+          - taglist.changed == true
+          - taglist.added_tags.Name == "task_definition-{{ resource_prefix }}"
+
```

Comment:
Also an idempotency check here

Reply:
Add twice.  On initial check both add_tags and tags.
On 2nd check tags and change == false